WalletWasabi/Stores/IndexStore.cs

```diff
@@ -71,13 +84,70 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
 		}
 	}
 
+	private void MigrateToSqliteNoLock()
+	{
+		int i = 0;
+
+		try
+		{
+			Logger.LogWarning("Migration of block filters to SQLite format is about to begin. Please wait a moment.");
+
+			Stopwatch stopwatch = Stopwatch.StartNew();
+
+			List<string> filters = new(capacity: 100_000);
+
+			using (FileStream fs = File.Open(OldIndexFilePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+			using (BufferedStream bs = new(fs))
+			using (StreamReader sr = new(bs))
+			{
+				while (true)
+				{
+					i++;
+					string? line = sr.ReadLine();
+
+					if (line is null)
+					{
+						break;
+					}
+
+					// Starting filter is already added at this point.
+					if (i == 1)
+					{
+						continue;
+					}
+
+					filters.Add(line);
+
+					if (i % 100_000 == 0)
+					{
+						IndexStorage.BulkAppend(filters);
+						filters.Clear();
+					}
+				}
+			}
+
+			IndexStorage.BulkAppend(filters);
+
+			Logger.LogInfo($"Migration of {i} filters to SQLite was finished in {stopwatch.Elapsed} seconds.");
+		}
+		catch (Exception ex)
+		{
+			Logger.LogError(ex);
+
+			SqliteConnection.ClearAllPools();
+
+			// Do not run migration code again if it fails.
+			File.Delete(NewIndexFilePath);
+			File.Delete(OldIndexFilePath);
+		}
+	}
+
 	/// <remarks>Guarded by <see cref="IndexLock"/>.</remarks>
 	private Task InitializeFiltersNoLockAsync(CancellationToken cancellationToken)
 	{
 		try
 		{
-			// TODO: Replace the message with a new one. Staying with this one to allow comparisons.
-			using IDisposable _ = BenchmarkLogger.Measure(LogLevel.Debug, "MatureIndexFileManager loading");
+			using IDisposable _ = BenchmarkLogger.Measure(LogLevel.Debug, "Block filters loading");
```

Comment:
Can we migrate here? This place seems like the proper place for initialization. If you put a long task here will it block the UI?

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/10551#issuecomment-1529734431